cmd/nvidia-ctk-installer/container/runtime/containerd/config_test.go

```diff
@@ -0,0 +1,978 @@
+
+      [plugins.cri.containerd.runtimes]
+
+        [plugins.cri.containerd.runtimes.nvidia]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.nvidia.options]
+            BinaryName = "/usr/bin/nvidia-container-runtime"
+            Runtime = "/usr/bin/nvidia-container-runtime"
+
+        [plugins.cri.containerd.runtimes.nvidia-cdi]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.nvidia-cdi.options]
+            BinaryName = "/usr/bin/nvidia-container-runtime.cdi"
+            Runtime = "/usr/bin/nvidia-container-runtime.cdi"
+
+        [plugins.cri.containerd.runtimes.nvidia-legacy]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.nvidia-legacy.options]
+            BinaryName = "/usr/bin/nvidia-container-runtime.legacy"
+            Runtime = "/usr/bin/nvidia-container-runtime.legacy"
+
+        [plugins.cri.containerd.runtimes.runc]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.runc.options]
+            Runtime = "/usr/bin/runc"
+`
+				require.Equal(t, expected, string(actual))
+				return nil
+			},
+			assertCleanupPostConditions: func(t *testing.T, testRoot string) error {
+				configPath := filepath.Join(testRoot, "etc/containerd/config.toml")
+				require.FileExists(t, configPath)
+
+				actual, err := os.ReadFile(configPath)
+				require.NoError(t, err)
+
+				expected := `version = 1
+
+[plugins]
+
+  [plugins.cri]
+
+    [plugins.cri.containerd]
+      default_runtime_name = "runc"
+      enable_cdi = true
+
+      [plugins.cri.containerd.runtimes]
+
+        [plugins.cri.containerd.runtimes.runc]
+          runtime_type = "io.containerd.runtime.v1.linux"
+
+          [plugins.cri.containerd.runtimes.runc.options]
+            Runtime = "/usr/bin/runc"
+`
+				require.Equal(t, expected, string(actual))
+				return nil
+			},
+		},
+		{
+			description:     "v1 legacy: existing config with default runtime",
+			configVersion:   1,
+			useLegacyConfig: true,
+			containerOptions: container.Options{
+				Config:         "{{ .testRoot }}/etc/containerd/config.toml",
+				RuntimeName:    "nvidia",
+				RuntimeDir:     "/usr/bin",
+				SetAsDefault:   true,
+				RestartMode:    "none",
+				ExecutablePath: "not-containerd",
+			},
+			options: Options{},
+			prepareEnvironment: func(t *testing.T, testRoot string) error {
+				configPath := filepath.Join(testRoot, "etc/containerd/config.toml")
+				require.NoError(t, os.MkdirAll(filepath.Dir(configPath), 0755))
+
+				// Write initial v1 legacy config with default runtime
+				initialConfig := `version = 1
+
+[plugins]
+  [plugins.cri]
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.default_runtime]
+        runtime_type = "io.containerd.runtime.v1.linux"
+
+        [plugins.cri.containerd.default_runtime.options]
+          Runtime = "/usr/bin/runc"
```

Comment:
In the context of v1 configs, there are two ways to specify the default runtime: 
1. as an explicit `default_runtime` section (which is deprecated)
2. as a reference to a defined runtime under `plugins.cri.containerd.default_runtime_name`

Even in the v1 config we don't handle the first case consistently (or we do to update the spec, but don't when reading the spec). Let's focus on 2. since this also matches how v2 configs are handled and allows us to further simplify things.